client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -1597,7 +1597,7 @@ private static String encodePart(String pathPart) {
                 //paths that start with `-` or contain `:`
                 URI uri = new URI(null, null, null, -1, "/" + pathPart, null, null);
                 //manually encode any slash that each part may contain
-                return uri.getRawPath().substring(1).replaceAll("/", "%2F");
+                return uri.getRawPath().substring(1).replaceAll("/", "%2F").replaceAll("\\+", "%2B");
```

Comment:
Why are we doing this manually and piecemeal instead of using `URLEncoder`?

Reply:
> "what needs to be encoded should already be encoded and that is why we create the URI instance to encode each part."

That cannot be accurate. If I encode the data before calling this, is going to be re-encoded.
So this code should be the responsible for encoding and it should happen only once. 

Why are you using URI ? and Not URLEncoder?